### PR TITLE
Add powerDynamic to Consumers

### PR DESCRIPTION
**Pull request by NiChrosia**
Adds an additional method to Consumers.java that allows a block to dynamically update power.

### CODE REVIEW

**Comment by NiChrosia** [comment]
(Doing that to minimize commits)

**Comment by NiChrosia** [comment]
Force push 2: Satellite commit is now gone.

File: core/src/mindustry/world/blocks/power/DynamicConsumePower.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package mindustry.world.blocks.power;
+
+import arc.func.*;
+import mindustry.gen.*;
+import mindustry.world.consumers.*;
+
+/** A power consumer that uses a dynamic amount of power, and activates sometimes */
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/power/DynamicConsumePower.java:7]:
```suggestion
/** A power consumer that uses a dynamic amount of power, and activates sometimes. */
```

File: core/src/mindustry/world/blocks/power/DynamicConsumePower.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package mindustry.world.blocks.power;
+
+import arc.func.*;
+import mindustry.gen.*;
+import mindustry.world.consumers.*;
+
+/** A power consumer that uses a dynamic amount of power, and activates sometimes. */
+public class DynamicConsumePower extends ConsumePower{
+    private final Floatf<Building> usage;
+    private final Boolf<Building> consume;
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/power/DynamicConsumePower.java:10]: The boolf seems redundant; you could just return 0 usage instead.

**Pull request merged**